packages/docusaurus-theme-classic/src/theme/CodeBlock/WordWrapButton/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useCallback, useState, useEffect} from 'react';
+import clsx from 'clsx';
+import {translate} from '@docusaurus/Translate';
+import type {Props} from '@theme/CodeBlock/WordWrapButton';
+
+import styles from './styles.module.css';
+
+export default function WordWrapButton({
+  codeBlockRef,
+  className,
+}: Props): JSX.Element | null {
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isCodeScrollable, setIsCodeScrollable] = useState<boolean>(false);
```

Comment:
Some random idea: should this setting be globally synced? We can also take advantage of the scroll blocker we have from the tab group syncing.

Reply:
I don't think so, what are the benefits exactly from this?

Reply:
That's the "random" part: I'm not sure 😄 For example, if I'm viewing an index page of all CLI commands, I'd want to toggle the wrapping of all code blocks so I can see all commands when scrolling through, rather than clicking them one after another.

Reply:
It's hard to say, I wouldn't do that for now.

Reply:
Makes sense, we can see